### ISSUE TITLE
Crashes on startup on High Sierra GM

### ISSUE THREAD

**Issue by simonpie** [HappenApps/Quiver#959]
It crashes on startup on High Sierra GM.

Here are the logs in `/var/log/system.log`
```Sep 16 10:50:23 Ananke com.apple.xpc.launchd[1] (com.happenapps.Quiver.1716[8135]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]```

**Comment by darthShadow** [comment]
Any updates? Looking forward to using Quiver on High Sierra.

**Comment by ylian** [comment]
@simonpie  This crash might not be related to High Sierra. I am running High Sierra myself and Quiver runs fine.

The crash might be due to a corrupted search index file. Try to delete this file:

```
~/Library/Containers/com.happenapps.Quiver/Data/Library/Application\ Support/Quiver/index.dat
```

Then open Quiver again.

**Comment by tmepple** [comment]
I am having what I think is the same crashing @simonpie mentioned on two separate Macs running High Sierra.  I've deleted the search index file but that's not helping.  I think it's crashing when trying to autosave while typing notes.  If you leave it running without editing notes it works fine.

`Sep 27 10:57:04 tme-imac-2014 Quiver[87395]: libcoreservices: __create_or_fix_relative_directory: 959: __dirhelper_create_relative returned (os/kern) failure`
`Sep 27 10:57:08 tme-imac-2014 com.apple.xpc.launchd[1] (com.apple.xpc.launchd.oneshot.0x1000000f.Quiver[87395]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]`

**Comment by ylian** [comment]
@tmepple Thanks. I think I have found the issue. Will push out a fix this week.

**Comment by ylian** [comment]
@tmepple Can you check if this build works? Thanks.
https://www.dropbox.com/s/7leynrvegqinw7w/Quiver%203.1%20Build%20112.zip?dl=0

**Comment by tmepple** [comment]
Hi @ylian, I just tried it out and unfortunately it's still crashing.  It takes longer to crash now while typing continuously... say ~45 seconds.  Previously it was crashing after ~10 seconds of typing.

`Sep 27 20:44:14 tme-mbp-2014 Quiver[41590]: libcoreservices: __create_or_fix_relative_directory: 959: __dirhelper_create_relative returned (os/kern) failure`
`Sep 27 20:44:20 tme-mbp-2014 com.apple.xpc.launchd[1] (com.happenapps.Quiver.16424[41590]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]`

**Comment by ylian** [comment]
@tmepple Can you try this one? Thanks.
 https://www.dropbox.com/s/wq6wvyybi8itjy9/Quiver%203.1%20Build%20113.zip?dl=0

**Comment by tmepple** [comment]
Thanks @ylian, that build is working great.  I have been editing notes continuously for the last 15 minutes with no crashes.  I will use the build all day tomorrow on both Macs and let you know if there are any crashes.

**Comment by ylian** [comment]
@tmepple Great! Thanks a lot for your help.

**Comment by tmepple** [comment]
Hi @ylian, just wanted to report I've used the new build 3.1/113 for the last 4 days and it works great w/zero crashing.  Thanks for getting it fixed so fast.

**Comment by simonpie** [comment]
@ylian  Hello.

Well, here `~/Library/Containers/com.happenapps.Quiver/Data/Library/Application\ Support/` was an alias for Quiver that was pointing to nowhere.  God knows what happened to it during High Sierra GM update.  Any way, I restaured from a time machine backup and now it seems to work well.

Tiny question, should I put the content of what i restored at the path you mentioned, or should I put alias at the path you mentioned and put the files somewhere else ?

Thank you for your answer.

**Comment by ylian** [comment]
@tmepple Thanks for letting me know.

**Comment by ylian** [comment]
@simonpie `~/Library/Containers/com.happenapps.Quiver/Data/Library/Application\ Support/` should be a real directory, not an alias. Not sure what happened during the update, but I'm glad you were able to restore the files.

Your Quiver library should be put at the default location `~/Library/Containers/com.happenapps.Quiver/Data/Library/Application\ Support/Quiver/Quiver.qvlibrary`. Don't use an alias.

**Comment by ylian** [comment]
Fixed in 3.1.1.

**Comment by jskye** [comment]
I don't use quiver and I also get this error (as well as other problems) ever since high Sierra security update. 
https://apple.stackexchange.com/questions/352645/update-screwing-with-encrypted-disk-or-keep-using-windows-boot
Seems to occur when a call to access a folder from a startup agent or Daemon but that folder doesn't exist. Even after I removed the rogue caller this error occurs which is strange.
